### PR TITLE
V2 ProgressText Implementation

### PR DESCRIPTION
**Pull request by praveen970**
Another variant of a Linear determinate Progress bar. ProgessText Implementation

### Validations
Auto tests

### Screenshots
image

### Pull request checklist

This PR has considered:
- [x] Light and Dark appearances
- [x] Automated Tests
- [x] Documentation and demo app examples
- [x] VoiceOver and Keyboard Accessibility
- [x] Internationalization and RTL layouts
- [ ] Size classes and window sizes (notched devices, multitasking, different window sizes, etc)

### CODE REVIEW

File: FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/demos/V2ProgressActivity.kt [hunk 1]
```diff
@@ -285,4 +293,44 @@ private fun IndeterminateProgressIndicatorExamples() {
...
+private fun ProgressTextDemo(linearProgress: Float, context: Context) {
+    Label(
+        modifier = Modifier.padding(top = 16.dp),
+        text = "Progress Text",
+        textStyle = FluentAliasTokens.TypographyTokens.Title2,
+        colorStyle = ColorStyle.Brand
+    )
+    Spacer(modifier = Modifier.height(12.dp))
+    ProgressText(
+        text = "Generating the summary...",
```
Comment (mishramayank1) [inline FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/demos/V2ProgressActivity.kt:308]: Also show text update wrt to Progress, Like "Started", "Completed"

File: fluentui_progress/src/main/java/com/microsoft/fluentui/tokenized/progress/ProgressText.kt [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+fun ProgressText(
+    text: String,
+    progress: Float,
+    onCancelClick: () -> Unit,
+    modifier: Modifier = Modifier,
+    progressTextTokens: ProgressTextTokens? = null
+) {
+    val tokens = progressTextTokens
+        ?: FluentTheme.controlTokens.tokens[ControlTokens.ControlType.ProgressText] as ProgressTextTokens
+    val progressTextInfo = ProgressTextInfo()
```
Comment (mishramayank1) [inline fluentui_progress/src/main/java/com/microsoft/fluentui/tokenized/progress/ProgressText.kt:52]: Take Progress as parameter in ProgressTextInfo()

File: fluentui_progress/src/main/java/com/microsoft/fluentui/tokenized/progress/ProgressText.kt [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+import androidx.compose.animation.core.LinearOutSlowInEasing
+import androidx.compose.animation.core.animateFloatAsState
+import androidx.compose.animation.core.tween
+import androidx.compose.foundation.*
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.text.BasicText
+import androidx.compose.material.Icon
```
Comment (mishramayank1) [inline fluentui_progress/src/main/java/com/microsoft/fluentui/tokenized/progress/ProgressText.kt:11]: Use Icon from Fluent library

File: fluentui_progress/src/main/java/com/microsoft/fluentui/tokenized/progress/ProgressText.kt (hunk 2 above)
Comment (NamanPandey) [reply fluentui_progress/src/main/java/com/microsoft/fluentui/tokenized/progress/ProgressText.kt:53]: Could be a good idea if colors are to be changed with respect to progress.

File: fluentui_progress/src/main/java/com/microsoft/fluentui/tokenized/progress/ProgressText.kt [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+                .border(
+                    border, borderColor, shape
+                )
+        ) {
+            Row(
+                modifier = Modifier.padding(padding),
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                Icon(
+                    ProgressTextIcons.DismissCircle,
```
Comment (NamanPandey) [inline fluentui_progress/src/main/java/com/microsoft/fluentui/tokenized/progress/ProgressText.kt:88]: I think the icon should be changeable as per partner requirement.

File: fluentui_progress/src/main/java/com/microsoft/fluentui/tokenized/progress/ProgressText.kt [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+                            role = Role.Button,
+                            onClick = onCancelClick
+                        ),
+                    tint = iconColor
+                )
+                Spacer(modifier = Modifier.width(iconTextSpacing))
+                BasicText(text = text, style = typography.merge(TextStyle(color = textColor)))
+            }
+        }
+        Box(modifier = Modifier.align(Alignment.BottomCenter)) {
```
Comment (NamanPandey) [inline fluentui_progress/src/main/java/com/microsoft/fluentui/tokenized/progress/ProgressText.kt:106]: 2 boxes in a Box with alignments set to orient them. Can we use Column or something? Any reason this hierarchy is used?

Comment (praveen970) [reply fluentui_progress/src/main/java/com/microsoft/fluentui/tokenized/progress/ProgressText.kt:106]: If we use Column, two boxes are set one below the other but the problem is we want the 2nd box to be on the bottom border of the parent box, so box is used with orientation bottomcenter

File: fluentui_progress/src/main/java/com/microsoft/fluentui/tokenized/progress/ProgressText.kt (hunk 4 above)
Comment (praveen970) [reply fluentui_progress/src/main/java/com/microsoft/fluentui/tokenized/progress/ProgressText.kt:88]: Removed oncancelclick and added parameter to take icon as a fluenticon. Onclick can be given through fluent icon

File: fluentui_progress/src/main/java/com/microsoft/fluentui/tokenized/progress/ProgressText.kt [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+    val progressIndicatorHeight = tokens.progressbarHeight(progressTextInfo = progressTextInfo)
+    val progressIndicatorBackgroundColor =
+        tokens.progressbarBackgroundColor(progressTextInfo = progressTextInfo)
+    val progressIndicatorColor = tokens.progressbarBrush(progressTextInfo = progressTextInfo)
+    val iconTextSpacing = tokens.iconTextSpacing(progressTextInfo = progressTextInfo)
+    val padding = tokens.padding(progressTextInfo = progressTextInfo)
+    val iconSize = tokens.iconSize(progressTextInfo = progressTextInfo)
+    val iconColor = tokens.iconColor(progressTextInfo = progressTextInfo)
+    val shape = RoundedCornerShape(12.dp)
+    Box(modifier = Modifier.clip(shape)) {
```
[the remark this answers is not in the document]
Comment (praveen970) [reply fluentui_progress/src/main/java/com/microsoft/fluentui/tokenized/progress/ProgressText.kt:73]: one box for the text and border, the other box will be aligned at bottomceter which is a bottom border of the first box

**Review by mishramayank1 — APPROVED** [review]

**Pull request merged**